### DISCUSSION TITLE
Specifying method "POST" in Request Class but GET gets used instead

### DISCUSSION THREAD

**Discussion by idealerror** [saloonphp/saloon#163] [Q&A] [answered]
Hello,

I'm trying to figure out why when I use `protected ?string $method = 'POST';` in a Request class that it actually calls a `GET`.  Is there any specific configuration changes needed in order to perform a `POST` over a `GET`? Please see below for my configuration of the Request and Connector with the Debug output.

Request:
```php
class AuthExplorerService extends SaloonRequest
{
    use CastsToDto;

    protected ?string $connector = Connector::class;
    protected ?string $response = CookieResponse::class;
    protected ?string $method = 'POST';

    /**
     * The endpoint of the request.
     *
     * @return string
     */
    public function defineEndpoint(): string
    {
        return '/finder';
    }

    protected function castToDto(SaloonResponse $response): object
    {
        return ExplorerTokenDto::fromSaloon($response);
    }
}
```

Connector:
```php
class Connector extends SaloonConnector
{
    use AcceptsJson;

    public function defineBaseUrl(): string
    {
        return 'https://www.myapi.com';
    }

    public function defaultHeaders(): array
    {
        return [
            'Content-Type' => 'application/json',
            'Accept' => 'application/json',
            'Accept-Language' => 'en-us',
            'User-Agent' => getUserAgent(),
        ];
    }

    public function defaultConfig(): array
    {
        return [
            'connect_timeout'  => 15,
            'timeout'          => 15,
            'read_timeout'     => 15,
            'verify'    => false,
            'debug' => true,
        ];
    }
}
```

Debug:
```
> GET /finder HTTP/1.1
Host: www.myapi.com
Content-Type: application/json
Accept: application/json
Accept-Language: en-us
User-Agent: MyUserAgent
```

**Comment by Sammyjo20** [comment, marked as answer]
Hey @idealerror this is really odd. I've just tested this locally and it seems to be working okay with the latest version of Saloon. What does your `getUserAgent()` method do? Could that be causing an error in Guzzle, meaning it errors when parsing the HTTP status code?

**Reply by idealerror** [reply]
`getUserAgent()` is a helper I created to pass the current version of the application in the header to the API.  There should be no issues there.

I didn't think about what version of Saloon I was running, but when you mentioned latest, I ran a `composer update`:
```bash
  - Upgrading sammyjo20/saloon (v1.6.0 => v1.6.2): Extracting archive
  - Upgrading sammyjo20/saloon-laravel (v1.6.0 => v1.7.0): Extracting archive
 ```

It's now working correctly.  I'm not sure if there were any changes to how methods were handled in the versions mentioned above, but everything seems to be working correctly now after updating. 

From debug:
```bash
> POST /finder HTTP/1.1
Host: www.myapi.com
Content-Type: application/json
Accept: application/json
Accept-Language: en-us
User-Agent: MyUserAgent
```

**Reply by Sammyjo20** [reply]
That's very strange. I'll do some tests on v1.6.0 thank you! Glad it's all sorted.
